superset/sql_lab.py

```diff
@@ -166,10 +165,8 @@ def execute_sql_statement(
         sql = parsed_query.as_create_table(query.tmp_table_name)
         query.select_as_cta_used = True
     if parsed_query.is_select():
-        if SQL_MAX_ROWS and (not query.limit or query.limit > SQL_MAX_ROWS):
-            query.limit = SQL_MAX_ROWS
-        if query.limit:
-            sql = database.apply_limit_to_sql(sql, query.limit)
+        # Enforce the correct limit generated at query creation time in sql_json
+        sql = database.apply_limit_to_sql(sql, query.limit)
```

Comment:
Are we sure that the `SQL_MAX_ROW` configuration is still taken into account and that the user cannot specify above it?

If people want a new "suggested" soft limit that the user can override, I'm supportive, but we still need an environment setting that the user cannot override.